### PR TITLE
Manifest fixes

### LINKED ISSUE #3119: Could not fetch frame data from the server (A manifest file not exists)
### My actions before raising this issue
- [x ] Read/searched [the docs](https://github.com/opencv/cvat/tree/master#documentation)
- [x ] Searched [past issues](/issues)

When I try to enter into any job in concrete task, I can't see any images and I get the following error:

```
Could not fetch frame data from the server
Error: . "\"cannot get requested data type: chunk, number: 0, quality: Quality.COMPRESSED\\nA manifest file not exists, index cannot be created\"".
``` 
task_issue

I've found the following logs in *cvat container* for this task:
```
[2021-04-22 11:14:26,412] ERROR cvat.server.task_52: cannot get requested data type: chunk, number: 0, quality: Quality.COMPRESSED
Traceback (most recent call last):
  File "/home/django/cvat/apps/engine/views.py", line 470, in data
    buff, mime_type = frame_provider.get_chunk(data_id, data_quality)
  File "/home/django/cvat/apps/engine/frame_provider.py", line 167, in get_chunk
    return self._loaders[quality].get_chunk_path(chunk_number, quality, self._db_data)
  File "/home/django/cvat/apps/engine/cache.py", line 29, in get_buff_mime
    chunk, tag = self.prepare_chunk_buff(db_data, quality, chunk_number)
  File "/home/django/cvat/apps/engine/cache.py", line 63, in prepare_chunk_buff
    reader = ImageDatasetManifestReader(manifest_path=db_data.get_manifest_path(),
  File "/home/django/cvat/apps/engine/media_extractors.py", line 361, in __init__
    self._manifest.init_index()
  File "/home/django/utils/dataset_manifest/core.py", line 287, in init_index
    self._index.create(self._manifest.path, 3 if self._manifest.TYPE == 'video' else 2)
  File "/home/django/utils/dataset_manifest/core.py", line 224, in create
    assert os.path.exists(manifest), 'A manifest file not exists, index cannot be created'
AssertionError: A manifest file not exists, index cannot be created
[2021-04-22 11:25:37,711] INFO cvat.server.task_52: get repository request
```
I've found that it is still possible to dump annotations but I can't export the full dataset. 

### Expected Behaviour
Images in the job should be loaded properly.

### Current Behaviour
Images are not loaded in the task with error notification.

### Context
Due to this issue I can't load the full dataset

### Your Environment

- Git hash commit (`git log -1`): 
```
commit be9e00fa7aeba432690901d54509760eb9ebfba4 (HEAD -> develop, origin/develop, origin/HEAD)
Author: Dmitry Kruchinin <33020454+dvkruchinin@users.noreply.github.com>
Date:   Wed Apr 21 14:47:58 2021 +0300

    Update cypress test. Canvas 3D functionality. Basic actions. (#3112)
    
    * Rename, add css class
    
    * Update cypress test.

```
- Docker version `docker version` (e.g. Docker 17.0.05): 20.10.05
- Are you using Docker Swarm or Kubernetes? `Nope`
- Operating System and version (e.g. Linux, Windows, MacOS): GNU/Linux Ubuntu 4.15.0-140-generic

### PR DESCRIPTION
**Pull request by Marishka17**
### Motivation and context

Resolves #3119 
- Fixed missed `force` flag for preparing video meta
- Fixed image filtering
- Added reverse function for migration

### How has this been tested?

In development environment, docker 
### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
~~- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly~~
- [ ] I have added tests to cover my changes
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
~~- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))~~

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
~~- [ ] I have updated the license header for each file (see an example below)~~

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by nmanovic** [comment]
@Marishka17 , could you please add a line into CHANGELOG.md?

**Review by azhavoro — APPROVED** [review]

File: cvat/apps/engine/migrations/0038_manifest.py [hunk 1]
```diff
@@ -1,48 +1,120 @@
...
 from django.db import migrations
 
 from cvat.apps.engine.models import (DimensionType, StorageChoice,
                                      StorageMethodChoice)
+from cvat.apps.engine.media_extractors import get_mime
 from utils.dataset_manifest import ImageManifestManager, VideoManifestManager
 
-def migrate_data(apps, shema_editor):
+def get_logger():
+    migration = os.path.basename(__file__).split(".")[0]
```
Comment (nmanovic) [inline cvat/apps/engine/migrations/0038_manifest.py:19]: Does splitext work here? But I'm fine with the solution as well.

File: cvat/apps/engine/migrations/0038_manifest.py [hunk 2]
```diff
@@ -1,48 +1,120 @@
...
+    logger = logging.getLogger(name=migration)
+    logger.setLevel(logging.INFO)
```
Comment (nmanovic) [inline cvat/apps/engine/migrations/0038_manifest.py:21]: We have a special file with log configs. Do you think we can put the code into https://github.com/openvinotoolkit/cvat/blob/develop/cvat/apps/engine/log.py?

File: cvat/apps/engine/migrations/0038_manifest.py [hunk 3]
```diff
@@ -1,48 +1,120 @@
...
+                    for idx, pts, _ in meta:
+                        meta_file.write(f"{idx} {pts}\n")
+            else:
+                name_format = "dummy_{}.txt"
+                sources = [db_image.path for db_image in db_data.images.all().order_by('frame')]
+                counter = itertools.count()
+                logger.info('Preparing of the dummy chunks has begun')
+                for idx, img_paths in itertools.groupby(sources, lambda x: next(counter) // db_data.chunk_size):
+                    if os.path.exists(os.path.join(upload_dir, name_format.format(idx))):
+                        logger.info(name_format.format(idx) + " alredy exists")
```
Comment (nmanovic) [inline cvat/apps/engine/migrations/0038_manifest.py:68]:
```suggestion
                        logger.info(name_format.format(idx) + " already exists")
```

File: cvat/apps/engine/migrations/0038_manifest.py [hunk 4]
```diff
@@ -1,48 +1,120 @@
...
-                    os.remove(os.path.join(upload_dir, 'meta_info.txt'))
+                os.remove(os.path.join(upload_dir, 'meta_info.txt'))
+                logger.info('{}/meta_info.txt has been deleted'.format(upload_dir))
             else:
                 for path in glob.glob(f'{upload_dir}/dummy_*.txt'):
                     os.remove(path)
+                    logger.info(f"{path} has been deleted")
             # it's necessary for case with long data migration
             if os.path.exists(os.path.join(upload_dir, 'manifest.jsonl')):
+                logger.info('Manifest file alredy exists')
```
Comment (nmanovic) [inline cvat/apps/engine/migrations/0038_manifest.py:94]:
```suggestion
                logger.info('Manifest file already exists')
```

File: cvat/apps/engine/migrations/0038_manifest.py [hunk 5]
```diff
@@ -1,48 +1,120 @@
...
+            if os.path.exists(os.path.join(upload_dir, 'manifest.jsonl')):
+                os.remove(os.path.join(upload_dir, 'manifest.jsonl'))
+                logger.info('A manifest file has been deleted')
+            if os.path.exists(os.path.join(upload_dir, 'index.json')):
+                os.remove(os.path.join(upload_dir, 'index.json'))
+                logger.info('A manifest index file has been deleted')
+            data_dir = upload_dir if db_data.storage == StorageChoice.LOCAL else settings.SHARE_ROOT
+            if hasattr(db_data, 'video'):
+                if os.path.exists(meta_path):
+                    logger.info('A meta_info.txt alredy exists')
```
Comment (nmanovic) [inline cvat/apps/engine/migrations/0038_manifest.py:52]:
```suggestion
                    logger.info('A meta_info.txt already exists')
```

**Comment by nmanovic** [comment]
@Marishka17 , I believe the PR is good. It is a good idea to use logging inside some migrations. Better to move the initialization code into a common place. Please fix and after that I'm OK with the PR. Great job!

**Comment by nmanovic** [comment]
I will merge the PR now to unblock release of CVAT 1.4.0. Please prepare a separate PR with improved logging.

**Review by nmanovic — APPROVED** [review]

**Pull request merged**